File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/HashUpdateStream.java
package com.onevgo.functions;

import java.io.IOException;
import java.io.InputStream;
import java.security.MessageDigest;

public class HashUpdateStream {
    public static int hashUpdateStream(MessageDigest context, InputStream handle) {
        int result = 0;
        try {
            byte[] b = new byte[8192];
            int len = -1;
            while ((len = handle.read(b)) != -1) {
                context.update(b, 0, len);
                result += len;
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
        return result;
    }

    public static void main(String[] args) {

    }
}

File: 01_Language/01_Functions/go/src/functions/hash_update_stream.go
package functions

import (
	"hash"
	"io"
)

func HashUpdateStream(context hash.Hash, handler io.Reader) {
	b := make([]byte, 8192)
	for true {
		n, err := handler.Read(b)
		if err != nil {
			break
		}
		context.Write(b[0:n])
	}
}
